### PR TITLE
[draft] Dagger Europa experiment

### PR DESCRIPTION
**Pull request by shykes**
Raw snapshot of experimenting with Dagger Europa API. Cleanup coming soon :)

### CODE REVIEW

**Comment by shykes** [comment]
@gerhard check out the last commit, I tweaked the structure from what we discussed, but I think it works.

To be continued :)

File: 2021/dagger/ci/ci.cue [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+					"env": env
+					"app": app
+					depsCache: "readonly"
+					buildCache: "locked"
+				}
+				workdir: "/app"
+				script: "mix do deps.compile, compile"
+                        },
+			// 5. Set image config
+			// FIXME: how does this actually work? Does it mutate the field? Or is the field somehow
```
[the remark this answers is not in the document]
Comment (helderco) [reply 2021/dagger/ci/ci.cue:235]: Here's a POC: https://cuelang.org/play/?id=W3DjhQhugO-#cue@export@cue

It starts to get a bit hairy when you need to drill down on struct values and lists, if you want to do it all in CUE.

**Comment by gerhard** [comment]
It has been tremendous seeing this come together, I intend to pick it up first thing after Christmas 🙌🏻

**Comment by shykes** [comment]
As discussed, I am continuing this [on the Dagger repo](https://github.com/dagger/dagger/pull/1227). The repository is not yet public, but if you are reading this and would like access to the tech preview, register on https://dagger.io with a mention of this PR, and we will fast track you :)

**Pull request closed without merging**